HELM-368: Update HELM docs (#624)
OpenNMS/opennms-helm: docs/modules/reference/pages/how-to-build-with-docker.adoc

Reviewer: Also, I'm not sure how this compares to the Docker section of the "Build from source" page.

> = Build with Docker

Author: ```suggestion
= Build from Source with Docker
```

This section describes how to build {page-component-title} with Docker.

== Simple build

> {page-component-title} can be compiled from source with a Docker build image:

Reviewer 2: ```suggestion
You can compile {page-component-title} from source with a Docker build image:
```

. Check out the repository, compile the code, and install the plugin:
+
[source, console]

> docker build -t myopennmshelmimage .

Reviewer 2: ```suggestion
docker build -t myopennmshelmimage 
```
not sure if the period was required. It looked odd.

Author: I think it might be required? I'll defer to others who are more familiar with console commands, though.

. Run the image:
+
[source, console]
docker run -p 3000:3000 -t myopennmshelmimage

. Log in to your {page-component-title} instance with the default admin credentials (`admin`/`admin`).
. Navigate to *Plugins* and enable the *OpenNMS Helm* plugin.

> IMPORTANT: The Grafana plugin directory location is changed from `/var/lib/grafana/plugins` to `/opt/grafana/plugins`.

Reviewer 2: ```suggestion
IMPORTANT: The Grafana plugin directory location changes from `/var/lib/grafana/plugins` to `/opt/grafana/plugins`.
```

Author: I think I want to check the tense and context on this - is it changed because of the steps the user takes, or because they're using Docker?

            --build-arg OPENNMS_HELM_GIT_BRANCH_REF=myBranch
----

This creates the plugin based on the GitHub URL and branch that you provide.

== Build service stack with Docker Compose

The {page-component-title} package includes an example `docker-compose.yml`, which builds a service stack with a compiled Grafana OpenNMS {page-component-title} plugin that you can test.
You can modify the build arguments in `docker-compose.yml` if you want to build from a specific GitHub fork, branch, or version tag.

When you first run `docker-compose up -d`, the Grafana {page-component-title} plugin is compiled from the source.
Subsequent runs will use the compiled image from your local system.
If you want to rebuild the image, run `docker-compose build --no-cache`.

The URL for your Horizon or Meridian datasource is `\http://opennms:8980/opennms`.

== Configure Docker for PDF reports

> Horizon and Meridian include three templates from which you can create PDF reports.

Author: ```suggestion
Horizon and Meridian include templates from which you can create PDF reports.
```

You can modify the build arguments in `docker-compose.yml` if you want to build from a specific GitHub fork, branch, or version tag.

When you first run `docker-compose up -d`, the Grafana {page-component-title} plugin is compiled from the source.
Subsequent runs will use the compiled image from your local system.
If you want to rebuild the image, run `docker-compose build --no-cache`.

The URL for your Horizon/Meridian datasource is `\http://opennms:8980/opennms`.

== Configure Docker for PDF reports

Horizon and Meridian include three templates from which you can create PDF reports.
They use an existing Grafana dashboard (see https://docs.opennms.com/horizon/31/operation/deep-dive/database-reports/templates/grafana.html[Grafana Dashboard] in the main OpenNMS documentation).
If you want to use this feature, and you are running {page-component-title} through Docker, you must complete additional configuration.
Refer to Grafana's https://grafana.com/docs/grafana/latest/administration/image_rendering/#remote-rendering-service[Remote rendering service documentation] for more information.

> Additional details are also available on https://github.com/grafana/grafana-image-renderer/blob/master/docs/remote_rendering_using_docker.md[GitHub].

Author: Looks like this document has been removed from current versions of the repo; I'll remove it for now and do some more investigation.

```suggestion
```
